lib/redlock/client.rb

```diff
@@ -83,6 +84,17 @@ class RedisInstance
           return 0
         end
       eos
+      # thanks to https://github.com/sbertrang/redis-distlock/blob/master/lib/Redis/DistLock.pm
+      # also https://github.com/sbertrang/redis-distlock/issues/2 which proposes the value-checking
+      EXTEND_SCRIPT = <<-eos
+        if redis.call( "get", KEYS[1] ) == ARGV[1] then
+          if redis.call( "set", KEYS[1], ARGV[1], "XX", "PX", ARGV[2] ) then
```

Comment:
@seamusabshere Thinking about it, unifying it into a single script sure changes the semantic of the algorithm, as "extending" could become a combination of "lock extended" (i.e. lock was still there with our `value`) and "new lock set" (key was expired). Not sure about the consequences though.